### PR TITLE
programmed-create-job-api-with-necessary-classes

### PR DESCRIPTION
**Pull request by mickey-dawson**
Programmed the CreateJob API with necessary Services, Controllers, Daos etc.

### CODE REVIEW

File: src/main/java/org/kainos/ea/DropwizardApplicationApplication.java [hunk 1]
```diff
@@ -18,14 +21,20 @@ public String getName() {
...
+                return configuration.getSwagger();
+            }
+        });
     }
 
     @Override
     public void run(final DropwizardApplicationConfiguration configuration,
                     final Environment environment) {
         // TODO: implement application
         environment.jersey().register(new TestController());
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/DropwizardApplicationApplication.java:36]: As you are adding a real controller you can remove `TestController`

File: src/main/java/org/kainos/ea/client/FailedToCreateJobException.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package org.kainos.ea.client;
+
+public class FailedToCreateJobException extends Throwable {
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/client/FailedToCreateJobException.java:3]: What should this exception do?

File: src/main/java/org/kainos/ea/db/jobDao.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package org.kainos.ea.db;
+import org.kainos.ea.cli.JobRequest;
+
+import java.sql.*;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/jobDao.java:4]: IntelliJ will have made this change, but can you make this not a wildcard import

File: src/main/java/org/kainos/ea/db/jobDao.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+public class jobDao {
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
+
+    public int createJob(JobRequest job) throws SQLException {
+        Connection c = databaseConnector.getConnection();
+
+        String insertStatement = "INSERT INTO JobRole (JobRoleName, Band, Responsibilities, Specification) VALUES (?,?,?,?)";
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/jobDao.java:12]: Minor thing, but can you add a `;` to the end of the query

File: src/main/java/org/kainos/ea/db/jobDao.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package org.kainos.ea.db;
+import org.kainos.ea.cli.JobRequest;
+
+import java.sql.*;
+
+public class jobDao {
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/jobDao.java:7]: The `DatabaseConnector` seems to be missing

File: src/main/java/org/kainos/ea/core/JobValidator.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package org.kainos.ea.core;
+
+import org.kainos.ea.cli.JobRequest;
+
+public class JobValidator {
```
[the remark this answers is not in the document]
Comment (mickey-dawson) [reply src/main/java/org/kainos/ea/core/JobValidator.java:4]: > Where have these constraints come from?

I imported it to allow me to use Job Request methods within the class. Think the issues should be resolved now

File: .gitignore [hunk 7]
```diff
@@ -33,4 +33,7 @@ build/
 .vscode/
 
 ### Mac OS ###
-.DS_Store
\ No newline at end of file
+.DS_Store
+db.properties
+db.properties
+db.properties
```
Comment (glenn-hamilton) [inline .gitignore:39]: Can remove the extra `db.properties` values from here

File: src/main/java/org/kainos/ea/core/JobValidator.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (glenn-hamilton) [reply src/main/java/org/kainos/ea/core/JobValidator.java:4]: How did you come up with that a Job Role shouldn't be more than 60 characters etc?

File: src/main/java/org/kainos/ea/db/DatabaseConnector.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class DatabaseConnector {
+    private static Connection conn;
+
+    public static Connection getConnection() throws SQLException {
+        String user, password, host, name;
+
+        if(conn != null && !conn.isClosed()) { return conn; }
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/DatabaseConnector.java:15]: Once you add linting, it might have an issue with this

**Pull request closed without merging**